runtime/dart_isolate.cc

```diff
@@ -314,25 +315,12 @@ bool DartIsolate::Initialize(Dart_Isolate dart_isolate) {
     return false;
   }
 
-  if (dart_isolate == nullptr) {
-    return false;
-  }
-
-  if (Dart_CurrentIsolate() != dart_isolate) {
-    return false;
-  }
+  FML_DCHECK(dart_isolate != nullptr);
+  FML_DCHECK(dart_isolate == Dart_CurrentIsolate());
 
   // After this point, isolate scopes can be safely used.
   SetIsolate(dart_isolate);
 
-  // We are entering a new scope (for the first time since initialization) and
-  // we want to restore the current scope to null when we exit out of this
-  // method. This balances the implicit Dart_EnterIsolate call made by
-  // Dart_CreateIsolateGroup (which calls the Initialize).
-  Dart_ExitIsolate();
-
-  tonic::DartIsolateScope scope(isolate());
-
   // For the root isolate set the "AppStartUp" as soon as the root isolate
   // has been initialized. This is to ensure that all the timeline events
   // that have the set user-tag will be listed user AppStartUp.
```

Comment:
From the code below do you have some sense as to which method might end up returning `false`?
None of the Dart_SetLibraryTagHandler, Dart_SetDeferredLoadHandler, UpdateThreadPoolNames can seem to do that. Only possibility seems to be `LogIfError` itself returning false, which happens if handle returned from dart api calls is neither UnhandledExceptionError, nor Error. I wonder what return value is, whether that should be fixed (too).

Reply:
It may not be this method that returns `false`. As mentioned in CL description we know (from crash stack trace) it's `DartIsolate::InitializeIsolate()` which returns `false`  - which looks like this:
```c++
bool DartIsolate::InitializeIsolate(...) {
  ...
  if (!embedder_isolate->Initialize(isolate)) { ... }

  if (!embedder_isolate->LoadLibraries()) {
    ... 
    return false;
  }
  ...
  return true;
}
```

My suspicion is that `LoadLibraries()` - which does all kinds of stuff, including calling into Dart code when setting up `dart:io` ... - fails. 

From https://github.com/flutter/flutter/issues/90478#issuecomment-924398181:
```
crash log:
...
I/scudo   (28282): Scudo OOM: The process has Exhausted 256M for size class 262160.
```
it indicates that this app is running in low-memory situations. So one option I can see is that when we initialize the libraries we get into a OOM situation.

Just a suspicion - really hard to verify. 

Though the bug itself is quite obvious when reading the code and I've verified that it fixes the error handling path.